txdbus/authentication.py

```diff
@@ -315,11 +320,13 @@ def _step_one(self, username, keyring_dir=None):
 
         self.challenge_str = binascii.hexlify(hashlib.sha1(
                                               os.urandom(8)).digest())
-        
-        msg = ' '.join( [self.cookieContext,
-                         str(self.cookieId),
-                         self.challenge_str] )
-        
+
+        msg = '{} {} {}'.format(
+            self.cookieContext,
+            self.cookieId,
+            self.challenge_str.decode(),
+        )
```

Comment:
Why not keep this binary (possible encoding-related changes may destroy the cookie after all) and use `b" ".join([…])` here instead?